### PR TITLE
feat: add metrics plugin

### PR DESCRIPTION
**Pull request by pedrobarco**

### CODE REVIEW

**Review by cbartolomeu — COMMENTED** [review]
Can you rebase this branch with master after the metrics plugin PR is merged? It has some changes that were there.

File: pkg/metric/dsplugin.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+	"time"
+
+	"github.com/graphbound/graphbound/pkg/httpds"
+)
+
+func NewHTTPDSPlugin() httpds.Plugin {
+	return func(c *httpds.Context) {
+		start := time.Now()
+		reqHost := c.Request.URL.Host
+		reqPath := c.FullPath()
```
Comment (cbartolomeu) [inline pkg/metric/dsplugin.go:14]: Doesn't the request have a path? Why don't we need a full path here?

File: pkg/metric/metric.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	}, []string{"method", "host", "path", "status"})
+	reqSz = promauto.NewSummary(prometheus.SummaryOpts{
+		Name: "http_request_length_bytes",
+		Help: "Content-Length of HTTP requests",
+	})
+	resSz = promauto.NewSummary(prometheus.SummaryOpts{
+		Name: "http_response_length_bytes",
+		Help: "Content-Length of HTTP responses",
+	})
+)
```
Comment (cbartolomeu) [inline pkg/metric/metric.go:27]: We're sharing these metrics between plugins. Shouldn't we have different metrics for each plugin, for example, have a `reqCnt` for a server and another for data sources?

File: pkg/metric/dsplugin.go (hunk 1 above)
Comment (pedrobarco) [reply pkg/metric/dsplugin.go:14]: Full path means the path with templated parameters. Similar to what happens in gin.
```go
fmt.Println(c.Request.URL.Path) // /users/12345
fmt.Println(c.FullPath()) // /users/:id
```

File: pkg/metric/metric.go (hunk 2 above)
Comment (pedrobarco) [reply pkg/metric/metric.go:27]: The metric already has dedicated labels for supporting multiple "hosts" so I guess there's no need to create a different metric. A metric should convey a way of measuring something and labels should differentiate the characteristics of the thing that is being measured ([docs](https://prometheus.io/docs/practices/naming/#labels)).

**Review by cbartolomeu — COMMENTED** [review]
Lets say I want to extend the metrics plugin to add one more metric apart from the ones the plugin already provides. Do I need to fork the plugin and add my metric or do I need to create another metric's plugin and use both plugins?

**Comment by pedrobarco** [comment]
> Lets say I want to extend the metrics plugin to add one more metric apart from the ones the plugin already provides. Do I need to fork the plugin and add my metric or do I need to create another metric's plugin and use both plugins?

You'll need to create a new plugin but you don't really need to use both plugins.
Using the core metrics plugin provides you with request metrics and a `/metrics` endpoint. If you don't need any of that you can drop it.

**Review by cbartolomeu — APPROVED** [review]

**Pull request merged**